reconcile/openshift_prometheusrules.py

```diff
@@ -0,0 +1,273 @@
+import json
+import semver
+import logging
+import traceback
+
+import utils.gql as gql
+import reconcile.openshift_base as ob
+import utils.jsonnet as jsonnet
+import utils.template as template
+
+from utils.defer import defer
+from utils.openshift_resource import (OpenshiftResource as OR)
+
+SLO_RULES = 'slo-rules.jsonnet.j2'
+
+PERFORMANCE_PARAMETERS_QUERY = """
+{
+  performance_parameters_v1 {
+    labels
+    name
+    component
+    prometheusLabels
+    namespace {
+      name
+      cluster {
+        observabilityNamespace {
+          name
+          cluster {
+            name
+            serverUrl
+            jumpHost {
+              hostname
+              knownHosts
+              user
+              port
+              identity {
+                path
+                field
+                format
+              }
+            }
+            automationToken {
+              path
+              field
+              format
+            }
+            disable {
+              integrations
+            }
+          }
+        }
+        name
+        serverUrl
+        jumpHost {
+          hostname
+          knownHosts
+          user
+          port
+          identity {
+            path
+            field
+            format
+          }
+        }
+        automationToken {
+          path
+          field
+          format
+        }
+        internal
+        disable {
+          integrations
+        }
+      }
+    }
+    sloRules {
+      name
+      kind
+      metric
+      percentile
+      selectors
+      httpStatusLabel
+    }
+    volume {
+      name
+      threshold
+      rule
+      additionalLabels
+    }
+    availability {
+      name
+      additionalLabels
+      rules {
+        latency
+        errors
+      }
+    }
+    latency {
+      name
+      threshold
+      rule
+      additionalLabels
+    }
+    errors {
+      name
+      target
+      rule
+      additionalLabels
+    }
+    rawRecording {
+      record
+      expr
+      labels
+    }
+  }
+}
+"""
+
+QONTRACT_INTEGRATION = 'openshift-prometheusrules'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+
+def labels_to_selectors(labels):
+    return ", ".join(['\'%s="%s"\'' % (k, v) for k, v in labels.items()])
+
+
+def build_rules_aoa(rules, category):
+    return " + ".join(['[%s__%s.rules]' % (r, category) for r in rules])
+
+
+def generate_resource(template_file, values):
+    template_env = template.get_package_environment()
+    tpl = template_env.get_template(template_file)
+    tpl.globals['labels_to_selectors'] = labels_to_selectors
+    tpl.globals['build_rules_aoa'] = build_rules_aoa
+    tpl.globals['load_json'] = json.loads
+    return OR(jsonnet.generate_object(tpl.render(values)),
```

Comment:
can we split this into more lines? that way if it fails we will be able to debug better.

something like:


```python
def generate_resource(template_file, values):
    template_env = template.get_package_environment()
    tpl = template_env.get_template(template_file)
    tpl.globals['labels_to_selectors'] = labels_to_selectors
    tpl.globals['build_rules_aoa'] = build_rules_aoa
    tpl.globals['load_json'] = json.loads

    rendered = tpl.render(values)
    jsonnet_resource = jsonnet.generate_object(rendered)

    return OR(jsonnet_resource,
              QONTRACT_INTEGRATION,
              QONTRACT_INTEGRATION_VERSION)
```

Reply:
sure, will do

Reply:
fixed in dfd5e96